### PR TITLE
[SMALLFIX] Access control exception cleanup

### PR DESCRIPTION
**Pull request by aaudiber**
- Use the PermissionDenied status exception over AccessControlException
- Improve some error messages
- Throw exceptions instead of returning empty strings when authentication is disabled but user/group are looked up

### CODE REVIEW

**Comment by aaudiber** [comment]
Jenkins, test this please

**Comment by aaudiber** [comment]
Jenkins, test this please

File: core/client/src/main/java/alluxio/client/file/options/CompleteUfsFileOptions.java [hunk 1]
```diff
@@ -37,15 +35,19 @@
...
-  public static CompleteUfsFileOptions defaults() throws IOException {
+  public static CompleteUfsFileOptions defaults() {
     return new CompleteUfsFileOptions();
   }
 
-  private CompleteUfsFileOptions() throws IOException {
-    mOwner = SecurityUtils.getOwnerFromLoginModule();
-    mGroup = SecurityUtils.getGroupFromLoginModule();
+  private CompleteUfsFileOptions() {
+    if (SecurityUtils.isSecurityEnabled()) {
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/file/options/CompleteUfsFileOptions.java]: would it make sense to push the if logic into the utility method? I would expect that this pattern will be true for every caller at which point it makes sense to encapsulate it

File: core/common/src/main/java/alluxio/security/authentication/AuthenticatedClientUser.java [hunk 2]
```diff
@@ -47,37 +45,30 @@ public static void set(String userName) {
...
-    try {
-      User user = get();
-      if (user == null) {
-        throw new AccessControlException(
-            ExceptionMessage.AUTHORIZED_CLIENT_USER_IS_NULL.getMessage());
-      }
-      return user.getName();
-    } catch (IOException e) {
-      throw new AccessControlException(ExceptionMessage.AUTHENTICATION_IS_NOT_ENABLED.getMessage());
+  public static String getClientUser() {
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/security/authentication/AuthenticatedClientUser.java]: minor: I think it makes more sense to call this method `getName`

File: core/common/src/main/java/alluxio/security/group/CachedGroupMapping.java [hunk 3]
```diff
@@ -36,6 +39,8 @@
  * implementation depends on {@link GroupMappingService}.
  */
 public class CachedGroupMapping implements GroupMappingService {
+  private static final Logger LOG = LoggerFactory.getLogger(CachedGroupMapping.class);
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/security/group/CachedGroupMapping.java]: is this needed for anything?

File: core/client/src/main/java/alluxio/client/file/options/CompleteUfsFileOptions.java (hunk 1 above)
Comment (aaudiber) [inline core/client/src/main/java/alluxio/client/file/options/CompleteUfsFileOptions.java]: done

File: core/common/src/main/java/alluxio/security/authentication/AuthenticatedClientUser.java (hunk 2 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/security/authentication/AuthenticatedClientUser.java]: changed

File: core/common/src/main/java/alluxio/security/group/CachedGroupMapping.java (hunk 3 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/security/group/CachedGroupMapping.java]: removed

**Comment by ifcharming** [comment]
LGTM, thanks for the cleanup!

**Comment by aaudiber** [comment]
Going to follow the more conventional approach of only using unchecked exceptions for coding errors, no need for this PR any more

**Pull request closed without merging**